### ISSUE TITLE
Docker Desktop Not Opening

### ISSUE THREAD

**Issue by vinaysharma14** [docker/for-mac#6456]
- [x] I have tried the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior
Docker Desktop should launch on Mac.

### Actual behavior
The Docker Desktop icon flashes on the dock briefly and it doesn’t launch. It crashes silently without any error messages on the screen. The `com.docker.diagnose` tool is unable to diagnose from the terminal, the output of which has been attached below.

### Information
  - macOS Version: 12.2.1
  - Intel chip or Apple chip: Intel
  - Docker Desktop Version: 4.11.1

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`
```bash
➜  ~ /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[SKIP] DD0028: is there available VM disk space?
[FAIL] DD0031: does the Docker API work? Cannot connect to the Docker daemon at unix://docker.raw.sock. Is the docker daemon running?
[1]    3128 killed     /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
```

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose gather --upload`

```bash
➜  ~ /Applications/Docker.app/Contents/MacOS/com.docker.diagnose gather --upload
Gathering diagnostics for ID 4DC1603E-1354-4CA3-BB31-4E0C839D67C2/20220819183640 into /var/folders/q1/9nwl1r8s4hgcp4q8_grx1kvmw56qkc/T/4DC1603E-1354-4CA3-BB31-4E0C839D67C2/20220819183640.zip.
This may take up to 15 minutes.
time="2022-08-20T00:06:40+05:30" level=info msg="Triggering Linux sysrq and log flushes via a unix socket: diagnosticd.sock" type=unixsock
time="2022-08-20T00:06:40+05:30" level=warning msg="/flush failed: Post \"http://unix/flush\": dial unix diagnosticd.sock: connect: no such file or directory" type=unixsock
[1]    3639 killed     /Applications/Docker.app/Contents/MacOS/com.docker.diagnose gather --upload
```

**Comment by cataglyphis** [comment]
same issue
- macOS 12.5.1
- M1
- docker desktop: 4.11.1

**Comment by fsparikshit** [comment]
Facing same issue.

macOS 12.5.1
M1
docker desktop: 4.11.1

**Comment by jeromelefeuvre** [comment]
Same issues. After reinstalling from scratch, got a `docker is damaged and can't be opened...`

**Comment by AndreHelwig** [comment]
Got exactly the same issue. 
- MacOS 12.5.1 
- Intel Mac
- Docker-desktop 4.11.1

**Comment by vaiso** [comment]
Same issue for me, including the `docker is damaged and can't be opened` @jeromelefeuvre mentioned above after reinstalling. 
Running on Mac OS Monterey 12.5.1, with an Intel chip and Docker desktop 4.11.1

Attempted fixes (which did not work):
* Kill all docker processes and restart machine
* Uninstall docker, delete every related folder (as per [this guide](https://www.drbuho.com/how-to/uninstall-docker-mac)), and then reinstall
* Run directly from install dmg volume - when I do this I get the error `com.docker.backend cannot start, exit code 9`
* Run diagnostics directly on the install dmg volume - can only run `check`. Running `gather -upload` results in the `Docker is damaged` error message.

**Comment by anikookar** [comment]
The same goes here, I couldn't open my docker desktop today.
A screenshot of the error:

<img width="579" alt="image" src="https://user-images.githubusercontent.com/111803720/186015591-eea6cc35-2a08-4575-84b8-9ab2ab26d284.png">

**Comment by pouellet** [comment]
One more here, same symptoms after a reboot of Docker.

* macOS 12.4 (21F79)
* M1 Max
* Docker 4.11.1

Diagnostics ID: 387E0361-02C8-4893-8120-B2A1139ABFCC/20220822211113

**Comment by gwpries** [comment]
I was able to get around the "docker is damaged" error with:

```xattr -d -r -s com.apple.quarantine /Applications/Docker.app```

Docker is still very busted, but it does a little bit more and will run the diagnostics without killing the process at that point.

**Comment by tech-consortium** [comment]
Same issue here.  Our entire development team is impacted.

**Comment by airforceguy** [comment]
Yesterday I upgraded to macOS 12.5.1 and docker desktop doesn't start up. Always end with error message “Docker.app” is damaged and can’t be opened. You should move it to the Trash.

**Info**

MacOS Version:  MacOS Monterey 12.2.1
Chip: Apple M1 Pro
Docker Desktop Version: 4.11

**Attempts**

- Cleanup and reinstall
- Re-Installed using brew
- Re-Installed using dmg
- Lowered the version to 4.9 Docker

Nothing works

**Comment by lit-in** [comment]
Could it be because Docker Desktop is [no longer free ](https://www.docker.com/blog/updating-product-subscriptions/) for larger organizations

**Comment by nikfarjam** [comment]
No, Even with a valid license Docker Desktop doesn't work after upgrade to MacOS Monterey 12.5.1

**Comment by subigre** [comment]
Same issue with same config Mac M1 and Monterey 12.5.1.
I tried with RD and it is also affected (multiple issues are reported in the GitHub repo)... the problem might be linked to the latest Montery update.

**Comment by jsantias** [comment]
I have the same issue with the same configurations. My current workaround is running docker from the attached drive after double clicking the `.dmg` file. Hope there's a fix soon

**Comment by ihornste** [comment]
Same issue, Hoping there is a fix soon...

**Comment by dvillarraga** [comment]
Same issues, 

MacOS 12.5.1
Intel Mac
Docker-desktop 4.11.1

😢

**Comment by ihornste** [comment]
This is the biggest on god no cap moment in history

**Comment by Ombar1** [comment]
Same here!!!

**Comment by dvalley** [comment]
Same here!

MacOS 12.4
Intel Mac
Docker-desktop 4.11.1

**Comment by kcem-flyr** [comment]
Please check your Antiviruses.

For example AMP Antivirus (based on clamav) is silently moving com.docker.backend and com.docker.driver.amd64-linux to quarantine and it is impossible to install/reinstall/open Docker.

Threat detected:
Osx.Exploit.CVE_2021_4034-9951522-2
Affected versions: all above 4.5.0.

Workaround: make exceptions in antivirus policy or use version 4.4.2 (the highest version for M1 chipset that is not detected as threat)

**Comment by AndreHelwig** [comment]
> Please check your Antiviruses.
> 
> For example AMP Antivirus (based on clamav) is silently moving com.docker.backend and com.docker.driver.amd64-linux to quarantine and it is impossible to install/reinstall/open Docker.

Good point. We are Using AMP as well and i will check if another rule can help to mitigate this issue.
Thanks for pointing there. Will provide a Update ASAP.

**Comment by gwpries** [comment]
> Please check your Antiviruses.
> 
> For example AMP Antivirus (based on clamav) is silently moving com.docker.backend and com.docker.driver.amd64-linux to quarantine and it is impossible to install/reinstall/open Docker.
> 
> Threat detected: Osx.Exploit.CVE_2021_4034-9951522-2 Affected versions: all above 4.5.0.
> 
> Workaround: make exceptions in antivirus policy or use version 4.4.2 (the highest version for M1 chipset that is not detected as threat)
> 
> <img alt="Zrzut ekranu 2022-08-23 o 12 11 30" width="1181" src="https://user-images.githubusercontent.com/97434712/186152512-0def0814-d1bf-41ba-9e86-ebec4808acb8.png">

We are also using AMP, adding a policy for /Applications/Docker.app has fixed this. (Update policy, delete Docker.app, re-install). Thanks!

**Comment by AndreHelwig** [comment]
It looks like as it is this checksum that causes the problems.

https://www.virustotal.com/gui/file/e70344c588419ce54e9c695e502255466432ce0f5599ce8f2302d6d308c51593

**Comment by nicks** [comment]
Looks like ClamAV may have fixed the problem, just saw it flip from "Detected" to "Undetected" on @AndreHelwig 's link.

**Comment by AndreHelwig** [comment]
> Looks like ClamAV may have fixed the problem, just saw it flip from "Detected" to "Undetected" on @AndreHelwig 's link.

You are right. Also reported the Problem to Docker directly. Seems to be a ClamAV problem. And should be fixed soon with signature updates. Thanks a lot for all your input regarding this issue.

as soon as @vinaysharma14  also confirm that it was a clamAV Signature problem for him i think we can Close the issue.

**Comment by petrpulc** [comment]
I can relay a confirmation that the ClamAV signatures are updated as we discuss the issue here. There are two other files that may be still affected...

**Comment by tech-consortium** [comment]
We can also confirm that we are seeing Cisco AMP quarantine the following docker files:

=================================================================================
| 2022-08-23 11:47 AM	| Detect				 | id: '13935716891713407'
| 			| 					 | threat: 'Osx.Exploit.CVE_2021_4034-9951522-2'
| 			| 					 | /private/tmp/d20220823-57132-1enfov8/Docker.app/Contents/MacOS/com.docker.driver.amd64-linux
| 2022-08-23 11:47 AM	| Quarantine				 | id: '13935716891713407'
| 			| 					 | threat: 'Osx.Exploit.CVE_2021_4034-9951522-2'
| 			| 					 | /private/tmp/d20220823-57132-1enfov8/Docker.app/Contents/MacOS/com.docker.driver.amd64-linux
| 2022-08-23 11:47 AM	| Detect				 | id: '13935716891713371'
| 			| 					 | threat: 'Osx.Exploit.CVE_2021_4034-9951522-2'
| 			| 					 | /private/tmp/d20220823-57132-1enfov8/Docker.app/Contents/MacOS/com.docker.backend
| 2022-08-23 11:47 AM	| Quarantine				 | id: '13935716891713371'


The Cisco AMP CLI reports the following ClamAV virus definitions are being used...

Definition Version:	  ClamAV(osx.cvd: 1261)
Definitions Published:	  osx.cvd: 22 Aug 2022 13-00 -0400
Definitions Last Updated: 2022-08-22 05:34 PM

Here are the SHA256 hashes of files it has detected...

  Detection id		 Name			 SHA256
=================================================================================
|13935716891713407	| /private/tmp/d20220823-57132-1enfov8/Docker.app/Contents/MacOS/com.docker.driver.amd64-linux
|		E70344C588419CE54E9C695E502255466432CE0F5599CE8F2302D6D308C51593
|13935716891713371	| /private/tmp/d20220823-57132-1enfov8/Docker.app/Contents/MacOS/com.docker.backend
|		68C4ADF03FCB0C223D606665818A679DBC3B8C07D53F9CFE269F292052465205
|13928055828022355	| /Applications/Docker.app/Contents/MacOS/com.docker.backend
|		03B6DF174C8E602254C2D67D3B5FA772E7C7E06EF12D5332824C82BAC00F94A1
|13928055828022348	| /Applications/Docker.app/Contents/MacOS/com.docker.driver.amd64-linux
|		A9CAD32254819E757002CDBDB8E238DB26619C48989E7411BE1F5B96AA7A1211

**Comment by airforceguy** [comment]
Docker is back in action. @AndreHelwig   thanks a ton.

**Comment by jeromelefeuvre** [comment]
Back for me too after upgrade of signatures and reinstall of docker! Thanks!!!!

**Comment by vinaysharma14** [comment]
> > Looks like ClamAV may have fixed the problem, just saw it flip from "Detected" to "Undetected" on @AndreHelwig 's link.
> 
> You are right. Also reported the Problem to Docker directly. Seems to be a ClamAV problem. And should be fixed soon with signature updates. Thanks a lot for all your input regarding this issue.
> 
> as soon as @vinaysharma14 also confirm that it was a clamAV Signature problem for him i think we can Close the issue.

@AndreHelwig I do not use [ClamAV](http://www.clamav.net/).

**Comment by dvillarraga** [comment]
@airforceguy @jeromelefeuvre how did you fix it guys? what did you do to uninstall and download the fixed version?

**Comment by airforceguy** [comment]
> @airforceguy @jeromelefeuvre how did you fix it guys? what did you do to uninstall and download the fixed version?

Reinstall worked.

**Comment by kcem-flyr** [comment]
Have you read the comments above? This issue was connected to ClamAv virus
definitions (used by AMP antivirus).

To resolve this problem try to update your antivirus definitions manually
and reinstall Docker (delete app and copy it again from package).

If this does not help it is new issue.

śr., 31 sie 2022, 03:33 użytkownik KEINOS ***@***.***>
napisał:

**Comment by nicks** [comment]
Hi all! Thanks for all the reports.

I'm going to close this issue. As noted above, most of the comments here are about the AMP/clamav issue, which is now resolved.

As an aside for folks on this thread still having problems: we usually don't provide support for Docker Desktop install / fail-to-start issues without either (1) clear repro steps or diagnostics, or (2) some reason to believe that there's a single widespread problem. The clamav issue is a good example of case (2). 

Because Docker Desktop has such a wide install base, we've found that many install issues tend to be finicky affects-only-one-machine problems that we can't realistically debug case by case.
